### PR TITLE
fix unless-stopped unexpected behavior

### LINKED ISSUE #35304: `restart=unless-stopped` not working as expected
**Description**
according to https://github.com/moby/moby/issues/11008, `restart=unless-stopped` is used to solve this issue, if container is stopped, it will not be restarted automatically after daemon restart.

while it is not working like that. ping @tonistiigi 
**Steps to reproduce the issue:**
1. docker run -tid --name test --restart unless-stopped busybox /bin/sh -c 'exit 1'
2. docker stop test
3. docker ps -- make sure container is exited
3. restart docker daemon 
4. docker ps -- you will see container get restarted again

**Describe the results you received:**
manually stopped restart=unless-stop container get restarted again after docker restart

**Describe the results you expected:**
container stay exited after daemon restart

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

Client:
 Version:      17.09.0-ce
 API version:  1.32
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:39:28 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.09.0-ce
 API version:  1.32 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:45:38 2017
 OS/Arch:      linux/amd64
 Experimental: false

**Output of `docker info`:**
not related

**Additional environment details (AWS, VirtualBox, physical, etc.):**
N/A

### PR DESCRIPTION
**Pull request by x1022as**
fix https://github.com/moby/moby/issues/35304 `restart=unless-stopped` not working as expected

Signed-off-by: Deng Guangxing <dengguangxing@huawei.com>

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by boaz0 — COMMENTED** [review]
We are not writing new tests to `integration-cli`.

**Comment by x1022as** [comment]
@ripcurld0  where should I add tests to?

**Comment by dnephin** [comment]
Is there maybe already a test case for this behaviour that can be fixed to catch the problem? This is a one line fix, so I wonder if it really needs an entirely new test case.

**Comment by x1022as** [comment]
@dnephin  yes, I can add this test to `TestDaemonRestartUnlessStopped` in integration-cli/docker_cli_daemon_test.go, is this acceptable?

**Comment by dnephin** [comment]
Yes, adding another container there sounds good

**Comment by x1022as** [comment]
Test case would fail occasionally, need some deep dig :(

**Comment by thaJeztah** [comment]
@x1022as whats the status on this one? Is this ready for review again?

**Comment by x1022as** [comment]
@thaJeztah sorry for late response.  
jenkins's passed,  ready for review:)

**Review by vdemeester — APPROVED** [review]
LGTM 🤔

File: daemon/start.go [hunk 1]
```diff
@@ -158,6 +158,7 @@ func (daemon *Daemon) containerStart(container *container.Container, checkpoint
 
 	if resetRestartManager {
 		container.ResetRestartManager(true)
+		container.HasBeenManuallyStopped = false
```
Comment (cpuguy83) [inline daemon/start.go]: What's the reasoning behind changing this?

Comment (x1022as) [reply daemon/start.go]: we don't know which one of `kill` and 'libcontainerd.ExitEvent' comes first.

`containerStart()` in `exitEvent` should not overwrite `HasBeenManuallyStopped ` in such case

**Comment by olljanat** [comment]
Derek add label: rebuild/*

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Pull request merged**